portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,18 +31,68 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  List<HashMap<String, String>> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> comments = new ArrayList<>();
-    comments.add("Hey !!! Just commenting - Prakash");
-    comments.add("Roses are red - Nirmala");
-    comments.add("Violets are blue - Sushant");
-    comments.add("What I am doing, I have no clue- Prashant");  
-    
+    response.setContentType("application/json");
+
     Gson gson = new Gson();
     String json = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String firstName = getParameter(request, "first-name", "");
+    String lastName = getParameter(request, "last-name", "");
+    String commentText = getParameter(request, "comment", "");
+    Date commentDate = new Date();
+
+    Entity commentEntity = new Entity("comment");
+    commentEntity.setProperty("firstName", firstName);
+    commentEntity.setProperty("lastName", lastName);
+    commentEntity.setProperty("commentText", commentText);
+    commentEntity.setProperty("date", commentDate);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    HashMap<String, String> commentsData = makeHashmapOfFields(firstName, lastName, commentText);
+
+    comments.add(commentsData);
+    
+    response.sendRedirect("/index.html#connect");
+  }
+
+/**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+/**
+   * @return  HashMap conatining the first name, last name, comment and date when comment
+   *           was made
+   */
+  private HashMap<String, String> makeHashmapOfFields(String firstName, String lastName, String comment) {
```

Comment:
It might be worthwhile to consider creating a container class to represent this information for a couple of reasons below:

1. A container class can help decouple how the information is stored vs how its used. Let's say we have a "Comment" class, then a caller will just be concerned with a Comment and how to use it. It won't matter whether Comment is backed by a List<HashMap>, 4 separate individual HashMaps, or something else.
2. We already have 4 properties (firstname, lastname, comment, date), but as they grow or change you'll need to add a parameter to this method, which could be unwieldy/hard to read.
3. This would make it easier to make certain properties optional, e.g. you only cared about firstname and lastname didn't matter. 
4. The reader/caller won't need read all the code to understand what the HashMap<> comprises of, whereas right now in order to get elements from the hashmap, one needs to know the key names, which are defined in the method. (e.g. 'comment' key here is different from 'commentText' property used above)

Let me know if this makes sense or if you feel I missed something, otherwise happy to discuss further :)